src/query/service/src/schedulers/fragments/plan_fragment.rs

```diff
@@ -194,6 +211,68 @@ impl PlanFragment {
         Ok(fragment_actions)
     }
 
+    fn reshuffle<T: Clone>(
```

Comment:
I think we can use this method in src/query/catalog/src/plan/partition.rs:reshuffle to reduce duplicate codes. But we can do this in the next pr.

Reply:
exception has been coverd in https://github.com/datafuselabs/databend/blob/main/tests/sqllogictests/suites/base/09_fuse_engine/09_0023_replace_into